cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1061,6 +1061,108 @@ TriangleMesh TriangleMesh::SelectFacesByMask(const core::Tensor &mask) const {
     return result;
 }
 
+// A helper to compute the vertex and triangle masks based on indices.
+// Additionally updates tris_cpu to new indices.
+template <typename intT>
+static void SBIUpdateMasksAndTrisCPUHelper(const core::Tensor &indices,
+                                           core::Tensor &vertex_mask,
+                                           core::Tensor &tris_mask,
+                                           core::Tensor &tris_cpu) {
+    const int64_t num_tris = tris_cpu.GetLength();
+    const int64_t num_verts = vertex_mask.GetLength();
+
+    // compute the vertices mask
+    intT *vertex_mask_ptr = vertex_mask.GetDataPtr<intT>();
+    const intT *indices_ptr = indices.GetDataPtr<intT>();
+    for (int64_t i = 0; i < indices.GetLength(); ++i) {
+        if (indices_ptr[i] >= num_verts) {
```

Comment:
just realised I need to check `< 0` too, and I assume that the vertex indices are consecutive w/o gaps in between.